File: src/main/java/me/miki/shindo/management/event/impl/EventRenderChunkPosition.java
package me.miki.shindo.management.event.impl;

import me.miki.shindo.management.event.Event;
import net.minecraft.client.renderer.chunk.RenderChunk;
import net.minecraft.util.BlockPos;

public class EventRenderChunkPosition extends Event {

    private final RenderChunk renderChunk;
    private BlockPos blockPos;

    public EventRenderChunkPosition(RenderChunk renderChunk, BlockPos blockPos) {
        this.renderChunk = renderChunk;
        this.blockPos = blockPos;
    }

    public BlockPos getBlockPos() {
        return blockPos;
    }

    public void setBlockPos(BlockPos blockPos) {
        this.blockPos = blockPos;
    }

    public RenderChunk getRenderChunk() {
        return renderChunk;
    }
}

File: src/main/kotlin/me/miki/shindo/management/event/impl/EventRenderChunkPosition.kt
package me.miki.shindo.management.event.impl

import me.miki.shindo.management.event.Event
import net.minecraft.client.renderer.chunk.RenderChunk
import net.minecraft.util.BlockPos

class EventRenderChunkPosition(
    private val renderChunk: RenderChunk,
    private var blockPos: BlockPos,
) : Event() {
    fun getBlockPos(): BlockPos = blockPos

    fun setBlockPos(blockPos: BlockPos) {
        this.blockPos = blockPos
    }

    fun getRenderChunk(): RenderChunk = renderChunk
}
